lilio/calendar.py

```diff
@@ -27,27 +21,173 @@
+    def gap_dateoffset(self):
+        return self._gap_dateoffset
+
+    def __repr__(self):
+        """String representation of the Interval class."""
+        props = [
+            ("role", self.role),
+            ("length", self.length),
+            ("gap", self.gap),
+        ]
+
+        propstr = ", ".join([f"{k}={repr(v)}" for k, v in props])
+        return f"{self.__class__.__name__}({propstr})"
+
+
+class Calendar:
+    """Build a calendar from scratch with basic construction elements."""
 
-    @abstractmethod
     def __init__(
         self,
         anchor: str,
-    ) -> None:
-        """For initializing calendars, the following five variables will be required."""
+        allow_overlap: bool = False,
+        mapping: Union[
+            None,
+            MappingYears,
+            MappingData,
+        ] = None,
+        intervals: Union[None, List[Interval]] = None,
+    ):
+        """Instantiate a basic container for building calendar using basic blocks.
+
+        This is a highly flexible calendar which allows the user to build their own
+        calendar with the basic building blocks of target and precursor periods.
+
+        Users have the freedom to create calendar with customized intervals, gap
+        between intervals, and even overlapped intervals. They need to manage the
+        calendar themselves.
+
+        Args:
+            anchor: String denoting the anchor date. The following inputs are valid:
+                    - "MM-DD" for a month and day. E.g. "12-31".
+                    - "MM" for only a month, e.g. "4" for March.
+                    - English names and abbreviations of months. E.g. "December" or "jan".
+                    - "Www" for a week number, e.g. "W05" for the fifth week of the year.
+                    - "Www-D" for a week number plus day of week. E.g. "W01-4" for the
+                        first thursday of the year.
+            allow_overlap: If overlapping intervals between years is allowed or not.
+                Default behaviour is False, which means that anchor years will be
+                skipped to avoid data being shared between anchor years.
+            mapping: Calendar mapping. Input in the form: ("years", 2000, 2020) or
+                ("data", pd.Timestamp("2000-01-01"), pd.Timestamp("2020-01-01")). The
+                calendar mapping is usually set with the `map_years` or `map_to_data`
+                methods.
+            intervals: A list of Interval objects that should be appended to the
+                calendar when it is initialized.
+
+        Example:
+            Instantiate a custom calendar and appending target/precursor periods.
+
+            >>> import lilio
+            >>> calendar = lilio.CustomCalendar(anchor="12-31")
+            >>> calendar
+            CustomCalendar(
+                anchor='12-31',
+                allow_overlap=False,
+                mapping=None,
+                intervals=None
+            )
+
+        """
```

Comment:
```suggestion
        Example:
            Instantiate a custom calendar and appending target/precursor periods.

            >>> import lilio
            >>> calendar = lilio.Calendar(anchor="12-31")
            >>> calendar
            Calendar(
                anchor='12-31',
                allow_overlap=False,
                mapping=None,
                intervals=None
            )

        """
```

Reply:
I missed this one, and I think `doctest` did not catch it. I'll have a look why 🤔

Reply:
OK I found out why. The auto API docs will **not** show documentation contained in a Class `__init__` method. If you want these to show up (and be tested) on the documentation, the docstring needs to be in the main docstring of the class.

However... If you want to have, e.g., suggestions while typing your code in a modern IDE, as soon as you type `Calendar(` you will be shown the `__init__` docstring. It would be nice to have the example show on both, but that means having duplicate docs.